src/main/java/shared/sharedObjects.java

```diff
@@ -0,0 +1,10 @@
+package shared;
+
+import data.Database;
+
+// Variables that need to be accessed between multiple class instances (i.e. Database initialization
+// variable).
+public class sharedObjects {
+  // public static
+  public static Database dataInstance = new Database("testDB");
```

Comment:
Is there a reason this needs to be shared between classes? It's not unusual to have multiple classes with their own `Database` objects referring to the same underlying database, and it's generally simpler.

Reply:
My reasoning here was that it's inefficient to construct a new database object, might as well share the object as we don't know how costly setting up the initial connection may be.

Reply:
Though I guess I would have thought this technique was simpler than having two places where you have to refer to the same database explicitly by name.

Reply:
My biggest concern is thread safety (though I'm not sure whether that's actually an issue in this particular case). We can expect multiple `Database` objects to coordinate their work behind the scenes, but if you _also_ want the `Database` object itself to be threadsafe, you've created a lot more work for yourself.

Re: inefficiency, I'd consider it premature optimization. The number of DB connections scales with number of server instances regardless, so you're only changing things by a constant factor. If the app starts up slowly with the naive implementation, I'd profile it to make sure that's actually where the bottleneck is.

Re: simplicity, it may indeed be simpler to have a single source of truth for the database name. But that should probably be a string constant, or a string in a config file, not necessarily an object. And I think it's simpler to reason about lifetimes when you keep as much stuff as possible local or explicitly passed in to your classes.

If it _is_ worthwhile to share a `Database` object, we would generally construct it in some initialization function that can pass it as a constructor parameter to the classes that need it. That tends to make testing easier. (But I'm not sure offhand how it would work with this framework.)

Reply:
The multithreaded point makes a lot of sense thanks for pointing that out I tend to overlook those concerns so I'm glad you brought it to my attention so I can be more aware of these issues in the future. I'll implement this change.

Reply:
Question about implementation of java servlets, is it true that the database instances scale with connections, or would one static instance be used through all instances? I assume its the former but I'm not actually certain about that?

Reply:
either way I'm swapping the implementation over

Reply:
I'm not sure I understand your question. My assertion above was that each server (not servlet) instance must connect to the DB at least once. I think that's true regardless of the servlet framework, because the servers are isolated from each other.

Reply:
I guess I was thinking it may be possible for each client connecting to our server to share the same database connection. My question was mainly in regard to whether a static variable defined the way I did was created for each server instance, or only the servlet itself and its variables are created for each connection.